diff --git a/DesignSystemContext.js b/DesignSystemContext.js
index df6418675532c4e5e7ec7cd96c863b10da2f96ab..425d3fe251e518aa800f011e6eb5f93c07a7f813 100644
--- a/DesignSystemContext.js
+++ b/DesignSystemContext.js
@@ -4,7 +4,7 @@ import { DesignSystemProvider as i } from "./DesignSystemProvider.js";
 import { useRootElement as n } from "./RootElementContext.js";
 function l(r) {
   const t = n();
-  return /* @__PURE__ */ e(o, { target: t, children: /* @__PURE__ */ e(i, { zIndexBase: 1e6, children: r.children }) });
+  return /* @__PURE__ */ e(o, { target: t, disableCSSOMInjection: true, children: /* @__PURE__ */ e(i, { zIndexBase: 1e6, children: r.children }) });
 }
 export {
   l as DesignSystemProvider
diff --git a/Modal.js b/Modal.js
index 0db985dc73d3868ba20539b70480313504d1a5ac..9897329b50dec792fa0108d9916f95d266968673 100644
--- a/Modal.js
+++ b/Modal.js
@@ -15,7 +15,7 @@ function P({
   const [r, m] = f(null), c = h(() => ({ onClose: n }), [n]);
   return x(() => {
     const e = document.createElement("div");
-    return e.dataset.testId = "CompactViewContainer", document.body.appendChild(e), m(e), () => {
+    return e.dataset.testId = "CompactViewContainer", e.style.cssText = "position:fixed;inset:0;z-index:999999;pointer-events:auto;", document.body.appendChild(e), m(e), () => {
       e.remove();
     };
   }, []), l ? /* @__PURE__ */ t(d.Provider, { value: c, children: r && /* @__PURE__ */ t(u, { children: v(
diff --git a/package.json b/package.json
index e52257bab1a177403566e7e653a7f1827b042844..54cb9b69b7ba60bce985a2f85c666190744cfa66 100644
--- a/package.json
+++ b/package.json
@@ -9,8 +9,8 @@
   "types": "index.d.ts",
   "dependencies": {},
   "peerDependencies": {
-    "react": ">=16.8.0 <19.0.0",
-    "react-dom": ">=16.8.0 <19.0.0"
+    "react": ">=16.8.0 <20.0.0",
+    "react-dom": ">=16.8.0 <20.0.0"
   },
   "devDependencies": {}
 }
